tests/test_dynamic_publisher_block.py

```diff
@@ -82,46 +82,46 @@ def test_reusing_pubs(self):
             self.assertEqual(pub.return_value.open.call_count, 1)
             pub.return_value.send.assert_called_with(signals)
 
-    def test_partitioning(self):
-        publisher = DynamicPublisher()
+    @patch(DynamicPublisher.__module__ + '.Publisher', side_effect=[Mock(), Mock(), Mock()])
```

Comment:
I don't think you need the `side_effect=[Mock()...Mock()]` part. Patches and mocks will already return Mock objects. You would only need to use side_effect if you wanted your method to return a specific Mock when it was called for the 2nd time, for example, or to raise exceptions.

Reply:
I tried that but when I did, it would return the _same_ mock for _each_ invocation.

What I really wanted here is to not peek into `block._cache` at all and patch the publisher to return a specific mock when called with `topic.foo` and `topic.bar` and `topic.baz`, and then assert the correct signal calls against each of those, but I couldn't figure out how to do that. I'll take a closer look at it now